src/components/encoding-pane/property-editor.tsx

```diff
@@ -14,26 +15,32 @@ export interface PropertyEditorProps extends ActionHandler<SpecEncodingAction> {
   fieldDef: ShelfFieldDef;
 }
 
-export class PropertyEditorBase extends React.PureComponent<PropertyEditorProps, {}> {
+export interface PropertyEditorState {
+  formData: any;
```

Comment:
The React-Form that we're using stores the formData result as arbitrary JSON, so there's no easy way to be type-safe in this situation, especially since the schema of the formData result can be very complex/convoluted.